src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -97,66 +103,23 @@ const DiffExprCompute = (props) => {
 
   // Returns true if each of the compared groups is made up of at least
   // 1 sample with more cells than a given minimum threshold.
-  const canRunDiffExpr = useCallback(() => {
-
-    if (selectedComparison === ComparisonType.WITHIN) return true;
-
-    if (!basis || !cellSet || !compareWith || !cellIdToSampleMap.length > 0) { return false; }
-
-    const basisCellSetKey = getCellSetKey(basis);
-
-    const cellSetKey = getCellSetKey(cellSet);
-    const compareWithKey = getCellSetKey(compareWith);
-
-    let basisCellIds = [];
-    if (basisCellSetKey === 'all') {
-      const allCellIds = sampleKeys.reduce((cumulativeCellIds, key) => {
-        const cellIds = properties[key].cellIds;
-        return cumulativeCellIds.concat(Array.from(cellIds));
-      }, []);
-      basisCellIds = new Set(allCellIds);
-    } else {
-      basisCellIds = properties[basisCellSetKey].cellIds;
-    }
-
-    const cellSetCellIds = Array.from(properties[cellSetKey].cellIds);
-
-    let compareWithCellIds = [];
-    if (['rest', 'background'].includes(compareWithKey)) {
-      const parentKey = getRootKey(cellSet);
-
-      const otherGroupKeys = hierarchy.find(obj => obj.key === parentKey)
-        .children.filter(child => child.key !== cellSetKey);
-
-      compareWithCellIds = otherGroupKeys.reduce((acc, child) => {
-        return acc.concat(Array.from(properties[child.key].cellIds));
-      }, []);
-    } else {
-      compareWithCellIds = Array.from(properties[compareWithKey].cellIds);
-    }
-
-    // Intersect the basis cell set with each group cell set
-    const filteredCellSetCellIds = cellSetCellIds.filter(cellId => basisCellIds.has(cellId));
-    const filteredCompareWithCellIds = compareWithCellIds.filter(cellId => basisCellIds.has(cellId));
-
-    const hasSampleWithEnoughCells = (cellSet) => {
-      // Prepare an array of length sampleIds to hold the number of cells for each sample
-      const numCellsPerSampleInCellSet = new Array(numSamples).fill(0);
 
-      // Count the number of cells in each sample and assign them into numCellsPerSampleInCellSet
-      cellSet
-        .forEach(cellId => {
-          const sampleIdx = cellIdToSampleMap[cellId];
-          numCellsPerSampleInCellSet[sampleIdx] += 1;
-        });
-
-      return numCellsPerSampleInCellSet.find(numCells => numCells > MIN_NUM_CELLS_IN_GROUP)
-    }
+  const canRunDiffExpr = useCallback(() => {
 
-    const cellSetHasSampleWithEnoughCells = hasSampleWithEnoughCells(filteredCellSetCellIds)
-    const compareWithHasSampleWithEnoughCells = hasSampleWithEnoughCells(filteredCompareWithCellIds)
+    if (selectedComparison === ComparisonType.WITHIN) return canRunDiffExprResults.TRUE;
+    if (!basis || !cellSet || !compareWith || !cellIdToSampleMap.length > 0) { return canRunDiffExprResults.FALSE; }
```

Comment:
We could also move this to checkCanRunDiffExpr

Reply:
moved